### ISSUE TITLE
Missing yo-doc for some zefops

### ISSUE THREAD

**Issue by ageron** [zefhub/zef#38]
The following zefops have no `yo` documentation:

* `yo(L)`
* `yo(allow_tombstone)`
* `yo(expect)`
* `yo(collect)`
* `yo(func)`

**Comment by zeyadkhaled** [comment]
- `L` is a relic from old traversal styles and should be removed altogether if there are no dependencies on it anymore.


- `allow_tombstone` and `expect` were both throwing an exception when yo was called on them, that has been fixed now, but they will remain for the time being without a docstring as they don't have any implementations but are used as flags, I think these should be revisited and changed to ValueTypes as they don't have any implementations. Would you agree @pengwyn?


- Both `collect` and `func` had an ugly repr and that has been fixed now.  As for showing a docstring, it would be tricky as they aren't exactly a ZefOp. But it is something I can play around with at some point.

**Comment by pengwyn** [comment]
I agree with making those `allow_tombstone` and `expect` turned into ValueTypes. Although `expect` I don't think has been used yet, so we might have to rejig that for a different purpose later on.
